File: langs/python-ddd/src/martian_ddd/domain/grid.py
from __future__ import annotations
from .types import Position


class Grid:
    def __init__(self, max_x: int, max_y: int) -> None:
        if max_x < 0 or max_y < 0:
            raise ValueError("Grid dimensions must be non-negative")
        self.max_x = max_x
        self.max_y = max_y

    def contains(self, pos: Position) -> bool:
        return 0 <= pos.x <= self.max_x and 0 <= pos.y <= self.max_y

File: langs/rust-ddd/martian-robot-ddd/src/domain/grid.rs
use super::position::Position;

#[derive(Copy, Clone, Debug)]
pub struct Grid {
    pub max_x: i32,
    pub max_y: i32,
}

impl Grid {
    pub fn new(max_x: i32, max_y: i32) -> Self {
        Self { max_x, max_y }
    }
    pub fn contains(&self, p: Position) -> bool {
        0 <= p.x && p.x <= self.max_x && 0 <= p.y && p.y <= self.max_y
    }
}
